api/v1alpha1/addon_types.go

```diff
@@ -58,7 +58,28 @@ const (
 	CCMProviderNutanix = "nutanix"
 )
 
-type Addons struct {
+type AWSAddons struct {
```

Comment:
this is a really interesting approach - i wonder if we should put other items in here down the line.

Reply:
Yes, I think we can use this to configure provider specific addons in future.